lib/grape/dsl/inside_route.rb

```diff
@@ -138,16 +138,20 @@ def status(status = nil)
         end
       end
 
-      # Set an individual header or retrieve
-      # all headers that have been set.
-      def header(key = nil, val = nil)
-        if key
-          val ? @header[key.to_s] = val : @header.delete(key.to_s)
-        else
-          @header
+      module HeaderMethods
```

Comment:
Should be in a separate file probably. And I don't love the name "HeaderMethods", maybe just "Headers"?